app/models/course/video.rb

```diff
@@ -135,10 +135,18 @@ def initialize_duplicate_tab(duplicator, other)
                end
   end
 
-  def url_unchanged
-    errors.add(:url, 'cannot be updated for videos with comments or watch data') if url_changed? &&
-                                                                                    persisted? &&
-                                                                                    url_unchangeble?
+  def changing_used_url?
+    url_changed? && persisted? && children_exist?
+  end
+
+  def destroy_children
```

Comment:
Might this have performance issues when deleting videos with many children? This feature will probably be rarely used and only when there are few children though so I'm okay to go with this first and change later if necessary.

Reply:
I only tested on small video. Will test on the bulkiest video we have on prod (id: 198) on Sunday when I get back.

Reply:
For worst video, the old implementation takes 3 minutes to destroy all children. The destroy statements are rewritten to avoid N+1 bye preloading all children and deleting from bottom up (events, then sessions, then submissions).